bsonutil/tags_test.go

```diff
@@ -2,52 +2,81 @@ package bsonutil
 
 import (
 	"testing"
-
 	"github.com/stretchr/testify/assert"
 )
 
 func TestTag(t *testing.T) {
 	assert := assert.New(t)
 
-	var err error
-	var tagVal string
-
-	// fetching the bson tag for a missing struct field should return an error
-
-	type sOne struct {
-	}
+	t.Run("Invalid", func(t *testing.T) {
+		var FieldOne string = `bson:"tag1"`
+		_, err := Tag(FieldOne, "")
+		assert.Error(err)
```

Comment:
nit: it's sort of an old thing to make a new assert variable, and we've moved away from this. 

It's no big deal, but it has the effect of attributing the failure of a test to the *testing.T instance passed to the outer function rather than the inner function. So just delete the assert := assert.New(t) and pass t to all of the assert calls.